Update to Alpine Linux 3.6

NathanZook opened the pull request:
This closes #52.  This code was checked by https://github.com/samsung-cnct/k2/pull/552

joejulian commented on imagerun.sh:
```diff
@@ -5,7 +5,8 @@ set -o pipefail
 apk add --no-cache  \
-    openssl \
+    libffi-dev \
+    openssl-dev \
```
These -dev packages will need removed in the cleanup at the bottom.

NathanZook replied:
If I remove the dev packages, won't that also remove the library subpackages and cause failures?  Especially with openssl, which was added previously and not removed.

joejulian replied:
The answer should be, "No." Removing the -dev packages should not remove any other packages and should only remove the -dev packages which have c headers and static libraries. If installing openssl-dev also installs openssl as a dependency, the remove should not remove the openssl dependency (according to the apk documentation).

If it does, explicitly re-adding openssl to the above `add` _should_ keep that from being uninstalled.

NathanZook replied:
I needed to add openssl to the list of installed packages.

NathanZook commented:
I did my take on https://github.com/samsung-cnct/k2-tools/pull/61.  Unfortunately, looking at the logs, https://common-jenkins.kubeme.io/job/K2/job/PR-552/9/console, it is clear that this did not suffice.  Before I pushed the image, I did a local. `up.sh`, `down.sh`, and `computed_kubectl`.

As for fixing Jenkins, I suspect that it is reading the Jenkins file from master.  There is a tantalizing configuration option under PR-552, but I did not want to play with it without at least trying to read up on it--and then talking to folks about what I found.

NathanZook commented:
Reading in more detail, it seems likely that I WAS using the correct image.  To demonstrate, I added a final ENV to the Dockerfile, and added a pipeline stage to print the variable.  It worked, so I would say that I've got strong evidence that the methodology is in fact correct.  See https://common-jenkins.kubeme.io/job/K2/view/change-requests/job/PR-552/12/consoleFull, lines 16 & 17.  Based on the successful run of https://common-jenkins.kubeme.io/job/K2/view/change-requests/job/PR-552/9/, and the evidence just mentioned that the methodology is sound, I would say that this is ready to merge.

coffeepac commented:
Great!  thanks for clarifying.  I think the reason #61 went south was due to a still-existing jenkins kubernetes pod template caching issue.  Its not biting now as it was being triggered by the now-fixed CI issues.
